internal/provider/resource_application_test.go

```diff
@@ -809,6 +1000,50 @@ func testAccResourceApplicationBasic_Minimal(modelName, charmName string) string
 		`, modelName, charmName)
 }
 
+func testAccResourceApplicationWithExistingModel(appName, username, modelName string) string {
```

Comment:
This does not test an existing model - you need to add a model data source to the plan.

Reply:
I create the model outside of Terraform for the reasons mentioned in the tests,
>// We do this outside of Terraform because we don't currently support
>// creating a model owned by a different user than the one making the connection.

Then test we can deploy an application into that model.